_posts/2018-03-16-CocoaPods-Metadata-Service.markdown

```diff
@@ -0,0 +1,93 @@
+---
+layout: post
+title:  "New Service: CocoaPods Metadata generation"
+author: orta
+categories: cocoadocs documentation cocoapods.org
+---
+
+This week we shipped a new behind-the-scenes service for CocoaPods authors: [cocoapods-metadata-service][cms] it handles
+a subset of CocoaDocs responsibilities but does not have the requirement of running on a Mac. This makes it trivial for
+us to maintain.
+
+It's been almost a year since [we announced that CocoaDocs][cd_down] was going to be shut down. CocoaDocs' sunsetting
+was intially slowed down by the team at [BuddyBuild][bb] offering to take over the project, but the migration wasn't
+finished before they were [acquired by Apple][appl]. This means they cannot take over community services.
+
+Instead I've started building out a simpler replacement that handles just the needs of the cocoapods.org website. Read
+on to find out what that looks like.
+
+<!-- more -->
+
+### Why build this?
+
+CocoaPods' web infrastructure needs to continue to be hardened. We have less contributors and a lot more people using
+CocoaPods. Moving more of our web services into heroku means we have less diversity in how projects are hosted get run.
+Hosting a Mac Mini on the internet is a lot of work, and I woudln't wish it on anyone else.
+
+There's a tricky thing here though. CocoaDocs would run through the full `pod install` process and use the files to
+generate a set of useful metrics for each pod:
+
+```sh
+install_size
+total_files
+total_comments
+total_lines_of_code
+doc_percent
+readme_complexity
+initial_commit_date
+rendered_readme_url
+license_short_name
+license_canonical_url
+total_test_expectations
+dominant_language
+builds_independently
+is_vendored_framework
+rendered_changelog_url
+rendered_summary
+spm_support
+```
+
+Not all of these attributes are feasible when running on heroku because it has an [ephemeral filesystem][eth]. For
+example a repo would include example projects and code which are removed by the `pod install` - CocoaDocs would then use
+the remaining files to generate details on. In order to work on a filesystem-less server, some of those metrics need to
```

Comment:
heroku has a FS though, it's just not preserved across dyno restarts

Reply:
true, yeah, I will re-write this